lib/ansible/runner/credential/machine_credential.rb

```diff
@@ -7,8 +7,10 @@ def self.auth_type
 
       def command_line
         {:user => auth.userid}.delete_blanks.merge(become_args).tap do |args|
-          # Add `--ask-pass` flag to ansible_playbook if we have a password to provide
+          # Add `--ask-pass` or `--ask-become-pass` flags to ansible_playbook
+          # if we have a password to provide
           args[:ask_pass] = nil if auth.password.present?
+          args[:ask_become_pass] = nil if auth.become_password.present?
```

Comment:
Shouldn't this still be conditional on `become_enabled` as well?
Not sure what would happen if we tell running to ask for a become password, but don't want to escalate privileges.

Reply:
I have no problem with that.  Might be excessive, but I don't see how it could hurt adding it.

Reply:
I guess my comment should have been reworded as a question:

What happens if we don't have `--become` but do provide `--ask-become-pass`?

Reply:
> What happens if we don't have `--become` but do provide `--ask-become-pass`?

Unsure, though I think I am going to have to test.

Regarding what I just said:

> I have no problem with that. Might be excessive, but I don't see how it could hurt adding it.

Might not be as simple, since we don't pass `:become_enabled` when building the `.command_line` in the individual credentials.  I think, to your question, it might be simpler to see what happens if this scenario comes up instead of trying add the code to cover this properly.  If it is an issue, we can go from there.

Reply:
Oh, and I will be the one testing this shortly, so the burden for testing that is on me.

Reply:
:+1: Sounds good

Yeah, it didn't seem to be an easy change.

Reply:
Sorry for the delay on this, but with the following `env/cmdline`:

```
--user vagrant --become-user vagrant --become-method sudo --ask-pass --ask-become-pass
```

It worked just fine.  The patch was in place for this one.